GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -2159,12 +2159,13 @@ private void QuickFetch()
             UICommands.RepoChangedNotifier.Notify();
         }
 
-        protected override bool ProcessCmdKey(ref Message msg, Keys keyData)
+        public override bool ProcessHotkey(Keys keyData)
         {
             return !IsDesignMode && HotkeysEnabled
-                && (base.ProcessCmdKey(ref msg, keyData) // generic handling of this form's hotkeys (upstream)
-                    || (!GitExtensionsControl.IsTextEditKey(keyData) // downstream (without keys for quick search)
-                        && (RevisionGridControl.ProcessHotkey(keyData)
+                && (base.ProcessHotkey(keyData) // generic handling of this form's hotkeys (upstream)
+                    || ((!GitExtensionsControl.IsTextEditKey(keyData) // downstream (without keys for quick search)
+                         || (keyData == (Keys.Control | Keys.A) && RevisionGridControl.ContainsFocus)) // route Ctrl+A away from RevisionGridControl
+                        && ((keyData != (Keys.Control | Keys.A) && RevisionGridControl.ProcessHotkey(keyData))
                             || (CommitInfoTabControl.SelectedTab == DiffTabPage && revisionDiff.ProcessHotkey(keyData))
                             || (CommitInfoTabControl.SelectedTab == TreeTabPage && fileTree.ProcessHotkey(keyData)))));
```

Comment:
Can we please break it in a series of easy to understand/maintain conditions?

Reply:
yes, done